### DISCUSSION TITLE
[playwright] More details on failed assertions and errors

### DISCUSSION THREAD

**Discussion by HugoImaios** [artilleryio/artillery#2784] [Artillery]
Hello !

Work done with #2677 is amazing (even more  with traces), kudos to Ines, but might get even better :
- for assertions errors, we could have the locator associated with the expectation 
- for other errors, maybe more detailed too, like the url when failing `waitForURL` calls

As I had a lot of `locator.click: Timeout 30000ms exceeded` with no other detail, I ended up setting different timeout values for different expects or actions, like `60001ms`  and `60002ms`, just as good old error codes, just as a workaround to identify clearly which `click` timedout :smile: but I don't think it's the best solution ! 

here is a call log from playwright output
```
Ge [Error]: Timed out 60005ms waiting for expect(locator).toBeVisible()

Locator: locator('.session-text')
Expected: visible
Received: hidden
Call log:
  - locator._expect with timeout 60005ms
  - waiting for locator('.session-text')
```
So I think playwright has the information we need on line 3 instead of only using the line 1
We could also use lines 7 and 8 as you can see

**Comment by hassy** [comment]
Thanks for the suggestion @HugoImaios! This is on our radar. Hoping to build something a bit more generic / useful for other things and not just Playwright stacktraces. :)

**Reply by HugoImaios** [reply]
If I happen to enhance [the cleanErrorMessage function](https://github.com/artilleryio/artillery/blob/784c9e4d1f381a217ab11ad789f69f7beff669b8/packages/artillery-engine-playwright/index.js#L380), would you like to take a look ?

**Reply by HugoImaios** [reply]
(and I look forward to the complete solution in the future obviously :eyes: )
